### PR TITLE
The PRs which are not approved run subset of tests 

### PR DESCRIPTION
**Pull request by potiuk**
This PR is an implementation of optimisation - to only run
default values for build matrix in case PR does not have
"Ready to Merge" label. Shortly this label will be automatically
set when the approval status of the PR is "OK" and last build
will be re-triggered in case it was a success.

This is a draft PR for now, we should not merge it yet, until
we have support for automation of labeling and re-triggering
the workflows.

It is based on #11820

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @tobked - prosze o approve :)

**Review by TobKed — APPROVED** [review]
approve for testing

File: .github/workflows/label_when_approved.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if: steps.selective-checks.outputs.run-tests
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          label: 'okay to test'
+          require_committers_approval: 'true'
+          # TODO: add comments and checks
+      - name: Label when approved by commiters for PRs that do not require tests
+        uses: TobKed/label-when-approved-action@master
+        id: label-non-test-prs-when-approved-by-commiters
+        if: ! steps.selective-checks.outputs.run-tests
```
Comment (TobKed) [inline .github/workflows/label_when_approved.yml:61]:
```suggestion
        if: !steps.selective-checks.outputs.run-tests
```

Comment (TobKed) [reply .github/workflows/label_when_approved.yml:61]: maybe this, I am not sure
```suggestion
        if: steps.selective-checks.outputs.run-tests != 'true'
```

Comment (potiuk) [reply .github/workflows/label_when_approved.yml:61]: Yep. I corrected it already. Can you disapprove/approve again please :)

**Review by TobKed — CHANGES_REQUESTED** [review]
request changes

**Review by TobKed — APPROVED** [review]

**Review by TobKed — APPROVED** [review]
approved again

**Review by higrys — CHANGES_REQUESTED** [review]
request changes

**Review by higrys — CHANGES_REQUESTED** [review]
Again

**Review by higrys — CHANGES_REQUESTED** [review]
again

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
Again 3

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
test 2

**Review by higrys — APPROVED** [review]
test3

**Review by higrys — CHANGES_REQUESTED** [review]
Test 4

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
test 5

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
test

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
aaaaaa

**Review by higrys — APPROVED** [review]

**Review by higrys — CHANGES_REQUESTED** [review]
test aaaaa

**Review by higrys — APPROVED** [review]

**Pull request closed without merging**